addons/account-switching/userscript.js

```diff
@@ -0,0 +1,111 @@
+const el = (el, properties = {}, children) => {
+  const element = document.createElement(el);
+  Object.keys(properties).forEach((key) => {
+    if (key === "listeners") {
+      Object.keys(properties.listeners).forEach((event) => {
+        element.addEventListener(event, properties.listeners[event]);
+      });
+    } else {
+      element[key] = properties[key];
+    }
+  });
+  if (Array.isArray(children)) {
+    children.forEach((child) => {
+      element.appendChild(child);
+    });
+  } else if (children) {
+    element.appendChild(children);
+  }
+  return element;
+};
+
+export default async function ({ addon, msg, global, console }) {
+  while (true) {
+    const dropdown = await addon.tab.waitForElement("ul.dropdown, ul.user-nav", { markAsSeen: true });
+    await addon.tab.waitForElement(".divider");
+    const username = await addon.auth.fetchUsername();
+
+    function sendBGMsg(type, username) {
+      return async function (event) {
+        if (!event.isTrusted) return;
+        addon.messaging.sendMessage({ type, username }, () => {
+          window.location.reload();
+        });
+      };
+    }
+
+    addon.messaging.sendMessage(
+      {
+        type: "get-accounts",
+      },
+      async (accounts) => {
+        let didDivider = false;
+        for (const account of accounts) {
+          if (account === username) continue;
+
+          const countReq = fetch(`https://api.scratch.mit.edu/users/${account}/messages/count`);
+          const { count } = await (await countReq).json();
+          const accountInfoReq = fetch(`https://api.scratch.mit.edu/users/${account}/`);
+          const {
+            profile: { images },
+          } = await (await accountInfoReq).json();
+
+          const elProps = {};
+          if (!didDivider) {
+            elProps.className = "divider";
+            didDivider = true;
+          }
+          dropdown.insertBefore(
+            el(
+              "li",
+              elProps,
+              el(
+                "a",
+                {
+                  href: "javascript:void(0)",
+                  listeners: {
```

Comment:
```suggestion
                {
                  href: "#",
                  listeners: {
```

Reply:
I think JavaScript:void let's the user know that some JavaScript is going to run instead of going to a local/different url. Plus why change the url when you can just run a void onclick.

Reply:
@TheColaber https://stackoverflow.com/a/1293130

Reply:
I don't really agree and think changing a url is worse but I think I'll just get rid of the href attribute altogether. Actually it was copilot that decided to do a JavaScript url, and I # intensionally didn't use a hash.